### PR TITLE
Added some extra repr to triton template buffers and added autotuned block configs to templated attention

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123335
* __->__ #123813
* #123768

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/select_algorithm.py [hunk 1]
```diff
@@ -1113,6 +1115,9 @@ def benchmark_in_current_process(choices):
                         if "illegal memory access" in msg:
                             msg += "\n\nEither error in template or triton bug.\n"
                         raise ErrorFromChoice(msg, choice, debug_str())  # noqa: TRY200
+                except OutOfResources as e:
+                    log.warning(e)
```
Comment (drisspg) [inline torch/_inductor/select_algorithm.py:1119]: Is there any helpful context you wanna add to this one?

**Review by drisspg — APPROVED** [review]
🏎️ would be interesting to run the benchmark with and without max autotune to compare the performance

**Review by shunting314 — APPROVED** [review]

**Comment by Chillee** [comment]
```
import torch
import torch
from torch import nn
import torch.nn.functional as F
from torch.nn.attention._templated_attention import _templated_attention
torch.set_default_device('cuda')

import triton
import triton.language as tl

torch.manual_seed(0)

def bench(f, name=None, iters=1000, warmup=5, display=True, profile=False):
    import time
    from triton.testing import do_bench

    for _ in range(warmup):
        f()
    if profile:
        with torch.profiler.profile() as prof:
            f()
        prof.export_chrome_trace(f"{name if name is not None else 'trace'}.json")

    us_per_iter = do_bench(lambda: f())*1000

    if name is None:
        res = us_per_iter
    else:
        res= f"{name}: {us_per_iter:.3f}us"

    if display:
        print(res)
    return res

torch._dynamo.config.cache_size_limit = int(1e9)

def f(q, k, v):
    def sin10(qk, *args):
        return qk
    return _templated_attention(q, k, v, score_mod=sin10)

def g(q, k, v):
    return F.scaled_dot_product_attention(q, k, v)

b = 8
h = 64
s = 4096
d = 128
q, k, v = torch.randn(b, h, s, d, dtype=torch.float16), torch.randn(b, h, s, d, dtype=torch.float16), torch.randn(b, h, s, d, dtype=torch.float16)
f = torch.compile(f, dynamic=False, mode="default")
bench(lambda: f(q, k, v))
torch._dynamo.reset()
f = torch.compile(f, dynamic=False, mode="max-autotune-no-cudagraphs")
bench(lambda: f(q, k, v))
bench(lambda: g(q, k, v))
```

```
default: 39116.19186401367
autotuned: 27146.14486694336
sdpa: 23343.78433227539
```

**Pull request closed without merging**